pkg/cmd/search/shared/searcher.go

```diff
@@ -0,0 +1,178 @@
+package shared
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"math"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/cli/cli/v2/pkg/search"
+)
+
+type searcher struct {
+	client *http.Client
+	host   string
+}
+
+func NewSearcher(client *http.Client, host string) *searcher {
+	return &searcher{
+		client: client,
+		host:   host,
+	}
+}
+
+func (s *searcher) Search(query search.Query) (search.Result, error) {
+	result := search.Result{}
+	path := fmt.Sprintf("https://api.%s/search/%s", s.host, query.Kind)
```

Comment:
Did you consider using `ghinstance.RESTPrefix` here instead constructing the hostname manually?